crates/sui-indexer-alt-framework/src/handlers/cp_mapping.rs

```diff
@@ -20,8 +20,61 @@ pub struct StoredCpMapping {
     pub epoch: i64,
 }
 
+pub struct PrunableRange {
+    from: StoredCpMapping,
+    to: StoredCpMapping,
+}
+
 pub struct CpMapping;
 
+impl PrunableRange {
+    /// Gets the tx and epoch mappings for both the start and end checkpoints.
+    ///
+    /// The values are expected to exist since the cp_mapping table must have enough information to
+    /// encompass the retention of other tables.
+    pub async fn get_range(
+        conn: &mut Connection<'_>,
+        from_cp: u64,
+        to_cp: u64,
+    ) -> QueryResult<PrunableRange> {
+        let results = cp_mapping::table
+            .select(StoredCpMapping::as_select())
+            .filter(cp_mapping::cp_sequence_number.eq_any([from_cp as i64, to_cp as i64]))
+            .order(cp_mapping::cp_sequence_number.asc())
+            .load::<StoredCpMapping>(conn)
+            .await?;
+
+        match results.as_slice() {
+            [first, .., last] => Ok(PrunableRange {
```

Comment:
This code assumes that `from_cp < to_cp` but we're not doing anything to confirm or check that assumption, which could introduce subtle bugs. Can we have a situation where the two checkpoints are equal? The function is completely public so there is nothing stopping someone passing all kinds of combination of `from_cp` and `to_cp` in here.